pkg/apis/serving/v1beta1/predictor_model.go

```diff
@@ -100,7 +100,12 @@ func (ss stringSet) contains(s string) bool {
 // GetSupportingRuntimes Get a list of ServingRuntimeSpecs that correspond to ServingRuntimes and ClusterServingRuntimes that
 // support the given model. If the `isMMS` argument is true, this function will only return ServingRuntimes that are
 // ModelMesh compatible, otherwise only single-model serving compatible runtimes will be returned.
-func (m *ModelSpec) GetSupportingRuntimes(cl client.Client, namespace string, isMMS bool) ([]v1alpha1.ServingRuntimeSpec, error) {
+func (m *ModelSpec) GetSupportingRuntimes(cl client.Client, namespace string, isMMS bool) (map[string]v1alpha1.ServingRuntimeSpec, error) {
+
+	modelProtcolVersion := constants.ProtocolV1
```

Comment:
I think we should still support the cases where a user:
  1. has a ServingRuntime that omits the `protocolVersions` list 
  2. deploys an InferenceService without specifying a `protocolVersion`

So, pretty much the current behavior we have now. Defaulting the modelProtocolVersion here to ProtocolV1 here will prevent the matching of the model to the intended ServingRuntime in these cases.

Thenmaybe when you have the final map of supported runtimes, first sort by protocol (to ensure v1 is at the top of the list), then by name or resourceversion) 

@yuzisun @njhill WDYT?

Reply:
yes @pvaneck . I agree.